swarm.go

```diff
@@ -181,6 +188,12 @@ func (s *Swarm) Process() goprocess.Process {
 }
 
 func (s *Swarm) addConn(tc transport.CapableConn, dir network.Direction) (*Conn, error) {
+	numOfPeers := len(s.conns.m)
+	nonZeroLimit := s.peerLimit > 0
+	if nonZeroLimit && numOfPeers >= int(s.peerLimit) && s.Connectedness(tc.RemotePeer()) != network.Connected {
+		tc.Close()
```

Comment:
Please log this.
Also, what's the purpose of the Connectedness check?

Reply:
Added in a log, the purpose of the Connectedness check is to continue accepting connections from already connected peers. We only reject connections from peers to which we have no active connections with.

Reply:
but you are not counting peers, you are counting connections.

Reply:
unless I am misunderstanding the code 
```go
len(s.conns.m)
```
wouldn't this represent all the peers with active connections currently ?

The type for `s.conns.m` is  
`map[peer.ID][]*Conn`

Reply:
oh that's true; sorry, my bad.